Libraries/Image/resolveAssetSource.js

```diff
@@ -87,26 +82,20 @@ function getPathOnDevserver(devServerUrl, asset) {
 }
 
 /**
- * Returns a path like 'assets/AwesomeModule'
- */
-function getBasePath(asset) {
-  // TODO(frantic): currently httpServerLocation is used both as
-  // path in http URL and path within IPA. Should we have zipArchiveLocation?
-  var path = asset.httpServerLocation;
-  if (path[0] === '/') {
-    path = path.substr(1);
-  }
-  return path;
-}
-
-/**
- * Returns a path like 'assets/AwesomeModule/icon@2x.png'
+ * Returns a path like 'assets/AwesomeModule/icon@2x.png' on iOS
+ * or 'drawable_ldpi/awesomemodule_icon.png' on Android
  */
 function getScaledAssetPath(asset) {
   var scale = pickScale(asset.scales, PixelRatio.get());
-  var scaleSuffix = scale === 1 ? '' : '@' + scale + 'x';
-  var assetDir = getBasePath(asset);
-  return assetDir + '/' + asset.name + scaleSuffix + '.' + asset.type;
+  if (Platform.OS === 'ios') {
```

Comment:
Every `Platform.OS` branch increases this module complexity be factor of 2x. Before your PR, the only place we check `Platform.OS` was `getPathInArchive`. Let's not add extra branching.

I think your code would be much easier to understand and support if you had 2 versions of `getScaledAssetPath` instead of one and only used platform check in `getPathInArchive`.